### PR TITLE
alts: de-flake AltsHandshakerStubTest

### LINKED ISSUE #4291: AltsHandshakerStubTest.closeEarlyTest is flaky
https://source.cloud.google.com/results/invocations/f0eef4f9-678f-47de-a55f-c5e40cb961dd
```
io.grpc.alts.internal.AltsHandshakerStubTest > closeEarlyTest FAILED
    java.lang.AssertionError: Unexpected message for java.io.IOException: Not true that <"Received an unexpected response."> contains <"Response stream closed">
        at io.grpc.alts.internal.AltsHandshakerStubTest.sendAndExpectComplete(AltsHandshakerStubTest.java:93)
        at io.grpc.alts.internal.AltsHandshakerStubTest.closeEarlyTest(AltsHandshakerStubTest.java:139)

        Caused by:
        java.io.IOException: Received an unexpected response.
            at io.grpc.alts.internal.AltsHandshakerStub.send(AltsHandshakerStub.java:60)
            at io.grpc.alts.internal.AltsHandshakerStubTest.sendAndExpectComplete(AltsHandshakerStubTest.java:90)
            ... 1 more
```

### PR DESCRIPTION
**Pull request by jyane**
Intended to resolve #4291.
Previously AltsHandshakerStubTest.closeEarlyTest is flaky because it is async.
De-flake the test by removing executors.

cc: @jiangtaoli2016, @ejona86.

### CODE REVIEW

**Comment by jiangtaoli2016** [comment]
Thanks much @jyane for your fix.

File: alts/src/test/java/io/grpc/alts/internal/AltsHandshakerStubTest.java [hunk 1]
```diff
@@ -139,7 +129,7 @@ public void closeEarlyTest() throws InterruptedException {
     sendAndExpectComplete();
   }
 
-  private class MockWriter implements StreamObserver<HandshakerReq> {
+  private static class MockWriter implements StreamObserver<HandshakerReq> {
     private StreamObserver<HandshakerResp> reader;
```
Comment (carl-mastrangelo) [inline alts/src/test/java/io/grpc/alts/internal/AltsHandshakerStubTest.java]: does this need to be volatile now?

Comment (jyane) [reply alts/src/test/java/io/grpc/alts/internal/AltsHandshakerStubTest.java]: @carl-mastrangelo friendly ping :)

**Comment by carl-mastrangelo** [comment]
@jyane sorry for the delay, thanks!

**Comment by jyane** [comment]
@carl-mastrangelo Thanks!

**Pull request merged**